### ISSUE TITLE
Does not make sense to accept *both* File-list and DirectoryEntry

### ISSUE THREAD

**Issue by sicking** [drufball/directory-download#2]
The explainer says

> Navigator.saveDirectory() could accept a list of File objects, a DirectoryEntry, or both.

It does not make sense to accept both a list of File objects and a DirectroyEntry at the same time. It's not even clear to me what syntax the "both" option intends to use. It is more correct to say "Navigator.saveDirectory() could accept a list of File objects, or a DirectoryEntry."

Alternatively, we could allow the list to contain both File objects and DirectoryEntry objects. In WebIDL terms, I would say that saveDirectory would use

```
partial interface Navigator {
  void saveDirectory((sequence<(File or DirectoryEntry)> or DirectoryEntry), optional DOMString nameHint);
};
```

If a sequence is _not_ used, and the nameHint argument is left out, it defaults to the name of the DirectoryEntry.

**Comment by drufball** [comment]
@sicking Good catch! I didn't realize I'd said they could be passed together. Do you think there's a specific ergonomics reason to expect both simultaneously? Otherwise it might be simpler to accept just one at a time.

**Comment by sicking** [comment]
I don't have any particular use cases in mind for accepting both File objects and DirectoryEntry objects at once.

It just seemed like a strange limitation to allow saving two files into a directory, but not saving two directories into a directory. And once you can accept lists of files, and lists of directories, it didn't seem harder to accept a combination of both.

But if that's harder to implement in Blink, or if it turns out to create weird edge cases that need to be specced, then we can certainly leave it out.

One thing that we should consider though is accepting FileEntry objects in addition to DirectoryEntry objects.

**Comment by smaug----** [comment]
DirectoryEntry is created currently only when user what DnD'ed directory to the page
(excluding Chrome only stuff). So why would DirectoryEntry ever used here.
Same applies to FileEntry. It isn't possible to create FileEntries (except via DnD).
